### ISSUE TITLE
Parsing of integers and floats calls back into Base parser

### ISSUE THREAD

**Issue by KristofferC** [ZacLN/CSTParser.jl#23]
Replacing the base parser to calls to CSTParser gives problem when parsing for example an integer or a float (see https://github.com/Keno/FancyDiagnostics.jl/issues/1).

I believe the reason is that https://github.com/ZacLN/CSTParser.jl/blob/b9a5f944bbd3c1de35ad3b1b5cbd7a0d3ddd4351/src/conversion.jl#L41 calls back into the Base parser with a string.

Shouldn't there exist a method like:

```
Expr(x::EXPR{LITERAL{Tokens.INTEGER}}) = Base.parse(Int, x.val)
```

and perhaps the same for floats?

**Comment by ZacLN** [comment]
Yup, though we need to introduce `Token` kinds for UInt, hex, oct etc. which is also required to fix https://github.com/ZacLN/CSTParser.jl/issues/22

**Comment by KristofferC** [comment]
I understand why a `UInt` `Token` is needed but not why any more than that are needed.

**Comment by ZacLN** [comment]
Right you are

**Comment by Keno** [comment]
Fixed in #24

**Comment by Keno** [comment]
This is fixed.
